aten/src/ATen/core/TensorImpl.h

```diff
@@ -229,6 +229,28 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
     return is_contiguous_;
   }
 
+  bool is_sparse() const {
```

Comment:
Footgun note: if you have a Variable, it will NEVER report true for any of these methods.

Reply:
Potential footgun note: at the moment, variables have the same TensorTypeId as their corresponding tensor, but if this ever changes, you have the potential to be very sad. Not actionable at the moment, I think.